slack_cancel.gs

```diff
@@ -131,4 +131,99 @@ function cancel(uniqueid, channelid, userid){
                                          'I\'ve notified the channel in the help request thread.');
   
-}
\ No newline at end of file
+}
+
+
+
+function cancel_su(row){
+  ///// COMMAND: /CANCEL_SU: cancel function as super-user (without the validity checks). This is used to cancel a request when its status is changed to "re-open" in the spreadsheet.
+  
+  
+  /// define variables
+  var globvar = globalVariables();
+  var mod_userid = globvar['MOD_USERID'];
+  var mention_requestCoord = globvar['MENTION_REQUESTCOORD'];
+  
+  var log_sheetname = globvar['LOG_SHEETNAME'];
+  var tracking_sheetname = globvar['TRACKING_SHEETNAME'];
+  
+  var webhook_chatPostMessage = globvar['WEBHOOK_CHATPOSTMESSAGE'];
+  var access_token = PropertiesService.getScriptProperties().getProperty('ACCESS_TOKEN'); // confidential Slack API authentication token
+  
+  var tracking_sheet_col_order = globvar['SHEET_COL_ORDER'];
+  var tracking_sheet_ncol = tracking_sheet_col_order.length;
+  var tracking_sheet_col_index = indexedObjectFromArray(tracking_sheet_col_order); // make associative object to easily get colindex from colname  
+  
+  var colindex_uniqueid = tracking_sheet_col_index['uniqueid'];
+  var colindex_channelid = tracking_sheet_col_index['channelid']; 
+  var colindex_phone = tracking_sheet_col_index['requesterContact'];
+  var colindex_status = tracking_sheet_col_index['requestStatus']; 
+  var colindex_volunteerName = tracking_sheet_col_index['slackVolunteerName']; 
+  var colindex_volunteerID = tracking_sheet_col_index['slackVolunteerID']; 
+  var colindex_slackts = tracking_sheet_col_index['slackTS']; 
+  var colindex_slacktsurl = tracking_sheet_col_index['slackURL'];
+  var colindex_channel = tracking_sheet_col_index['channel']; 
+  var colindex_requestername = tracking_sheet_col_index['requesterName'];
+  var colindex_address = tracking_sheet_col_index['requesterAddr'];
+  
+  
+  
+  // load sheet
+  var spreadsheet = SpreadsheetApp.getActiveSpreadsheet();
+  var sheet = spreadsheet.getSheetByName(tracking_sheetname);
+  
+  // find requested row in sheet  
+  var rowvalues = sheet.getRange(row, 1, 1, tracking_sheet_ncol).getValues()[0];
+  
+  // store fields as separate variables
+  var uniqueid = rowvalues[colindex_uniqueid];
+  var channelid = rowvalues[colindex_channelid]; 
+  var requesterName = rowvalues[colindex_requestername];
+  var address = rowvalues[colindex_address];
+  var slackurl = rowvalues[colindex_slacktsurl];
+  var status = rowvalues[colindex_status]; 
+  var volunteerID = rowvalues[colindex_volunteerID];
+  var slack_ts = rowvalues[colindex_slackts];
+    
+
+  // reply to slack thread to confirm volunteer sign-up (chat.postMessage method)
+  var out_message = '<!channel> <@' + volunteerID + '> is no longer available. Can anyone else volunteer? Type `/volunteer ' + uniqueid + '`.';
```

Comment:
Consider moving this logic to a separate function so that it can be reused across cancel functions.